### PR TITLE
Bug/GAT-595

### PR DESCRIPTION
**Pull request by f3401pal**
return empty publish URL when in HEAD branch so that the uploadArchives will compile during gradle:sync but does NOT allow publish

### CODE REVIEW

**Review by johnm7115 — COMMENTED** [review]
It looks like temporary branches won't build.

File: src/test/groovy/LibraryNameTest.groovy [hunk 1]
```diff
@@ -232,4 +232,18 @@ class LibraryNameTest extends Specification {
         where:
         baseVersion << ['1', 'MOCOM-353', 'Test.Version', '1.Something.0', '1.2.3.4']
     }
+
+    @Unroll
+    def "Library version is correct for arbitrary commit in HEAD branch: #HEAD"() {
```
Comment (jackeys) [inline src/test/groovy/LibraryNameTest.groovy]: You don't need @Unroll and #HEAD here. That's only for parameterized tests (i.e. tests that have a where: section at the end).

File: src/test/groovy/PublishUrlTest.groovy [hunk 2]
```diff
@@ -58,6 +58,18 @@ class PublishUrlTest extends Specification {
         getPublishUrlRunner().buildAndFail()
     }
 
+    @Unroll
+    def "Publish URL is empty for HEAD branch on build server: #HEAD"() {
```
Comment (jackeys) [inline src/test/groovy/PublishUrlTest.groovy]: You don't need @Unroll and #HEAD here. That's only for parameterized tests (i.e. tests that have a where: section at the end).

**Review by jackeys — CHANGES_REQUESTED** [review]
This looks much better. Remove the unnecessary unrolls in your tests and you're good to go.

**Comment by f3401pal** [comment]
Temporary branches will build. There are tests for that. But `getLibraryVersion` will fail.
I think we should make `getLibraryVersion` fail on HEAD branch as well.

**Comment by jackeys** [comment]
Failing branches that don't begin with a JIRA key was intentional, which is why temporary branches will fail. Maybe the build script isn't the best way to enforce that?

File: lib_common.gradle [hunk 3]
```diff
@@ -34,8 +34,6 @@ abstract class LibraryVersionFormatter {
                 switch (branch) {
                     case "develop":
                         return new DevelopVersionFormatter()
-                    case "HEAD":
-                        return new HeadVersionFormatter()
```
Comment (jackeys) [inline lib_common.gradle]: This was necessary to allow the library to build on a HEAD commit, which was the point of the bug. Without it, you may be able to do a Gradle sync, but you can't build, which breaks rebase --exec, bisect, etc.

You can test this by using your build script locally in mocom-sdk. I just did the following:

```diff
-apply from: "https://www.github.com/geotracsystems/android-build-scripts/raw/1.3/lib_common.gradle"
+apply from: "/home/jkeys/Development/in-house/android/libraries/build-scripts/lib_common.gradle"

**Review by jackeys — CHANGES_REQUESTED** [review]
The last commit undid the point of the bugfix, unfortunately.

File: lib_common.gradle (hunk 3 above)
Comment (f3401pal) [reply lib_common.gradle]: Yes. I have tested it. Because `getPublishUrl` returns empty URL in HEAD branch, `fromBranch`is not reached in this case.

**Review by jackeys — APPROVED** [review]

**Pull request merged**